### PR TITLE
provider: create resources to add permissions to the server

### PR DESCRIPTION
**Pull request by iamniting**
create role, roleBinding, serviceAccount to grant permission to the API
server for accessing the other resources.

Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: controllers/storagecluster/provider_server.go [hunk 1]
```diff
@@ -279,6 +316,61 @@ func GetProviderAPIServerSecret(instance *ocsv1.StorageCluster) *corev1.Secret {
 	}
 }
 
+func GetProviderAPIServerPermissionResources(instance *ocsv1.StorageCluster) (*rbacv1.Role, *rbacv1.RoleBinding, *corev1.ServiceAccount) {
+
+	role := &rbacv1.Role{
```
Comment (sp98) [inline controllers/storagecluster/provider_server.go:321]: why do we need to create these resources in golang code and not use .yaml files like other resources?

Comment (iamniting) [reply controllers/storagecluster/provider_server.go]: done

**Review by sp98 — APPROVED** [review]

File: deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -1435,7 +1435,7 @@ metadata:
       \                   \"resources\": {}\n                }\n            ]\n        }\n
       \   }\n\t"
     operatorframework.io/suggested-namespace: openshift-storage
-    operators.operatorframework.io/builder: operator-sdk-v1.2.0
+    operators.operatorframework.io/builder: operator-sdk-v1.15.0+git
```
Comment (nb-ohad) [inline deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:1438]: Are we going backword in operator-sdk versions?

Comment (iamniting) [reply deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:1438]: No this is latest one

File: hack/source-manifests.sh [hunk 3]
```diff
@@ -89,7 +89,7 @@ function gen_ocs_csv() {
 	pushd config/manager
 	$KUSTOMIZE edit set image ocs-dev/ocs-operator="$OCS_IMAGE"
 	popd
-	$KUSTOMIZE build config/manifests | $OPERATOR_SDK generate bundle -q --overwrite=false --version "$CSV_VERSION"
+	$KUSTOMIZE build config/manifests | $OPERATOR_SDK generate bundle -q --overwrite=false --version "$CSV_VERSION" --extra-service-accounts ocs-provider-server
```
Comment (umangachapagain) [inline hack/source-manifests.sh:92]: Let's use a variable to define extra serviceaccounts. It'll be easier to add more accounts later.
If no extra accounts are provided, this arg can be skipped.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Review by umangachapagain — CHANGES_REQUESTED** [review]
Fix the typo in third commit message.

File: hack/source-manifests.sh (hunk 3 above)
Comment (iamniting) [reply hack/source-manifests.sh]: made changes

**Comment by iamniting** [comment]
> Fix the typo in third commit message.

done

File: deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (umangachapagain) [reply deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:1438]: We can only use sdk v1.13.1 or older due to K8s deps.

**Review by umangachapagain — CHANGES_REQUESTED** [review]

**Review by umangachapagain — APPROVED** [review]

**Comment by iamniting** [comment]
/hold 

for testing

**Review by umangachapagain — APPROVED** [review]

**Review by nb-ohad — APPROVED** [review]

**Pull request merged**